src/Aspire.Dashboard/ResourceService/Partials.cs

```diff
@@ -93,7 +93,7 @@ ImmutableArray<UrlViewModel> GetUrls()
             return (from u in Urls
                     let parsedUri = Uri.TryCreate(u.FullUrl, UriKind.Absolute, out var uri) ? uri : null
                     where parsedUri != null
-                    select new UrlViewModel(u.Name, parsedUri, u.IsInternal))
+                    select new UrlViewModel(u.Name, parsedUri, u.IsInternal, u.DisplayProperties is not null ? new UrlDisplayPropertiesViewModel(u.DisplayProperties.HasDisplayName ? u.DisplayProperties.DisplayName : null, u.DisplayProperties.HasSortOrder ? u.DisplayProperties.SortOrder : null) : null))
```

Comment:
I think we can simplify things here. A UrlViewModel should always have UrlDisplayPropertiesViewModel. Create one from the values available.

Something like:
```suggestion
                    select new UrlViewModel(u.Name, parsedUri, u.IsInternal, new UrlDisplayPropertiesViewModel((u.DisplayProperties?.HasDisplayName ?? false) ? u.DisplayProperties.DisplayName : null, (u.DisplayProperties?.HasSortOrder ?? false) ? u.DisplayProperties.SortOrder : null)))
```

Then remove logic that worries about null display properties VM from other code.

Reply:
I think it would be simpler to not have the concept of null at all in the view model.
* Empty string display name = no display name
* 0 and null sort orders are already equivelent so just have sort order of `int` and default to 0.